pkg/acp/envoy.go

```diff
@@ -144,3 +151,19 @@ func (w *EnvoyWatcher) IsAlive() bool {
 func (w *EnvoyWatcher) IsReady() bool {
 	return w.IsAlive()
 }
+
+func getDefaultReadyURL() string {
+	var readyPort uint64
+	var err error
+	strReadyPort := os.Getenv("AMBASSADOR_READY_PORT")
+	if strReadyPort != "" {
+		readyPort, err = strconv.ParseUint(strReadyPort, 10, 15)
+		if err != nil {
+			dlog.Infof(context.Background(), "Unable to parse AMBASSADOR_READY_PORT or port is out of bounds: %s", err)
+		}
+	}
+	if readyPort < 1 {
+		readyPort = 8006
+	}
+	return fmt.Sprintf("http://localhost:%d/ready", readyPort)
```

Comment:
I think we are fine with `localhost` here based on this https://cs.opensource.google/go/go/+/refs/tags/go1.19.4:src/net/ipsock.go;drc=38cfb3be9d486833456276777155980d1ec0823e;l=310.

@kflynn - Do you know if Golang will correctly resolve localhost to an ipv6 address, when in ipv6 only?

Reply:
The resolver should do the right thing with `localhost`, always forever and for true – if not, that's a serious bug.